tsl/test/expected/transparent_decompression_ordered_index-12.out

```diff
@@ -145,8 +145,8 @@ ORDER BY c.id;
 ---------------------------------------------------------------------------------------------------------------------------------------------------------
  Limit (actual rows=10 loops=1)
    ->  Custom Scan (DecompressChunk) on _hyper_3_11_chunk (actual rows=10 loops=1)
-         ->  Sort (actual rows=1 loops=1)
-               Sort Key: compress_hyper_4_12_chunk._ts_meta_sequence_num DESC
+         ->  Sort (actual rows=2 loops=1)
+               Sort Key: compress_hyper_4_12_chunk._ts_meta_max_1 DESC
```

Comment:
This looks like a regression to me. There is no need to do batch merging here as the query constraints only select one segment and we can use the compress_orderby without merging.

Reply:
I updated the PR. The optimization is now only chosen if we can't push down the sort and consume the tuples in this order.